### ISSUE TITLE
Yamlcreate.ps1 throws exception: "Join-Path : A positional parameter cannot be found that accepts argument"

### ISSUE THREAD

**Issue by philnach** [microsoft/winget-pkgs#3061]
# Brief description of your issue
Ran the YamlCreate.ps1 script from directly from the tools directory results in invalid argument exception when the tool goes to build manifest due to a package id folder not existing

# Steps to reproduce

Run YamlCreate.ps1 directly from the tools directory.  Below are the specified values I entered:

Id: skylot.jadx
Version: 1.1.0
Name: jadx
Publisher: skylot
License: Apache License 2.0
LicenseUrl: https://github.com/skylot/jadx/blob/master/LICENSE
AppMoniker: jadx
Tags: Java, Dex, android, APK
Description: Command line and GUI tools to produce Java source code from Android Dex and APK files
Homepage: https://github.com/skylot/jadx
Arch: Neutral
Url: https://github.com/skylot/jadx/releases/download/v1.1.0/jadx-gui-1.1.0-no-jre.exe
Sha256: D2D8FBB71BAEC7FFE99633231397DFF419AB12D09C18171E75DAE6D994C79984
InstallerType: exe

# Expected behavior

YamlCreate.ps1 doesn't throw an exception and creates manifest in a package id folder

# Actual behavior

Exception is thrown during script execution:
```
Join-Path : A positional parameter cannot be found that accepts argument 'System.Object[]'.
At C:\repos\winget-pkgs\Tools\YamlCreate.ps1:247 char:21
+     $FileLocation = Join-Path "\" "manifests" $id.Split('.')[0..1]
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.JoinPathCommand
```

# Environment

Windows 10 2004 OS Build 19041.450

```none
[winget --info]
Windows: Windows.Desktop v10.0.19041.450
Package: Microsoft.DesktopAppInstaller v1.10.42101.0
```

**Comment by perplexityjeff** [comment]
I have tried to reproduce the issue but I am unable to do so. I did however use PowerShell Core 7. I did also use x86 as the arch.

![testjadx](https://user-images.githubusercontent.com/19505844/90122610-d7c60f80-dd5d-11ea-957b-12949d69212d.png)

![image](https://user-images.githubusercontent.com/19505844/90122761-165bca00-dd5e-11ea-8ee1-15f960a4c182.png)

**Comment by denelon** [comment]
@philnach are you able to reproduce the issue?
If so, which version of PowerShell/Terminal are you using?

**Comment by philnach** [comment]
@denelon, I'll attach the PowerShell execution transcript for you later today.

**Comment by philnach** [comment]
@denelon , I tried with x86 as well, still the same failure.  Attached is the PowerShell transcript
[YamlCreateTranscript.log](https://github.com/microsoft/winget-pkgs/files/5075508/YamlCreateTranscript.log)

**Comment by philnach** [comment]
@denelon, I think I figured it out.  

I'm on PowerShell 5.1 and in PowerShell 6.0 "[-AdditionalChildPath](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/join-path?view=powershell-6#parameters)" was added which let's you append multiple paths together [StackOverflow suggestion](https://stackoverflow.com/a/53745818).  We could fix the line by doing the following:

$FileLocation = Join-Path "\" -ChildPath "manifests" | Join-Path -ChildPath $id.Split('.')[0] | Join-Path -ChildPath $id.Split('.')[1]

I'm more than happy to create a PR, unless you think we should require a minimum of PowerShell 6 for running the script?

**Comment by denelon** [comment]
@philnach as long as this isn't going to be a breaking change for other versions of PowerShell, I'm fine with supporting multiple earlier versions.

**Comment by philnach** [comment]
@denelon, perfect, I'll get the change coded up and verify PowerShell 5.1, 6.0 and 7.  Probably send out a PR tomorrow (8/21).

**Comment by megamorf** [comment]
@philnach If you're going to submit a PR please don't submit a fix that looks like this

    $FileLocation = Join-Path "" -ChildPath "manifests" | Join-Path -ChildPath $id.Split('.')[0] | Join-Path -ChildPath $id.Split('.')[1]

That code that is hard to maintain and does not follow PowerShell best practices. It probably makes sense to add a minimum [version requirement](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_requires?view=powershell-7) to the script. This can be done by beginning the `YamlCreate.ps1` script with the line:

```powershell
#Requires -Version 5.1
```

You can then use PSScriptAnalyzer to tell you about incompatible version specific features.

**Comment by philnach** [comment]
@megamorf, thanks for the feedback, I'm not an expert when it comes to PowerShell, it would be awesome if you could fix the script to work with PowerShell 5.1?  I had something slightly different, but it's good you brought up that it doesn't follow best practice.
